addons/pkg/util/tkr_util.go

```diff
@@ -90,3 +104,36 @@ func GetBootstrapPackageNameFromTKR(ctx context.Context, clt client.Client, pkgR
 
 	return "", pkgNamePrefix, fmt.Errorf("no bootstrap package prefixed with '%s' is found in the TKR object", pkgNamePrefix)
 }
+
+func FetchTKRSpecFromAnnotations(tkc map[string]string) (*runtanzuv1alpha3.TanzuKubernetesRelease, error) {
+	controlPlaneTKR := &runtanzuv1alpha3.TanzuKubernetesRelease{}
+
+	tkrSpec, ok := tkc["run.tanzu.vmware.com/tkr-spec"]
```

Comment:
"run.tanzu.vmware.com/tkr-spec" should be defined as a constant in addons/pkg/constants/constants.go